Add page on validator payout mechanism (fixes #234)

laboon opened the pull request:
Fixes #234 

This adds a page with a brief description and some examples of how the validator payout mechanism works and some examples showing why it is encouraged (and beneficial) to run multiple validators.

laboon commented:
@lsaether @jplatts12 Can you review?

laboon commented:
Not sure why circleCI can't even hit the main page.  Did something change recently with the setup?  I'm rebased against latest master.

lsaether commented on docs/polkadot/node/guides/validator-payout.md:
```diff
@@ -0,0 +1,70 @@
+
+Note that this is different than most other proof-of-stake systems such as Cosmos.  As long as a validator is in the validator set, it does not matter how much stake is behind that particular validator, in terms of rewards.  Validator `v1`, with 18 DOTs staked, received the same reward (2 DOTs) in this era as `v4` who had only 7 DOTs staked.
+
+The initial goal of the Kusama network is to have 50 - 100 validators after the initial proof-of-authority phase.  Not all of these validators must be run by different entities (companies or people); in fact, it is often to your advantage to run multiple validators.
```
```suggestion
The initial goal of the Kusama network is to have 50 - 100 validators after the initial proof-of-authority phase.  Not all of these validators must be run by different entities (companies or people); in fact, it may be to your advantage to run multiple validators.
```

I would avoid making a judgement on this until we see what goes down in the wild.

laboon replied:
Yeah, this was part of the original rationale for the issue but you're right, I don't think it needs to be made explicit and Jack is fine with removing it as well.

laboon replied:
Removed

lsaether commented on docs/polkadot/node/guides/validator-payout.md:
```diff
@@ -0,0 +1,70 @@
+
+## Running Multiple Validators
+
+It is possible for a single entity to run multiple validators.  It is explicitly encouraged by the protocol, as some examples will show.  Assuming you have enough DOTs, or enough stake nominates your validator, to ensure that your validators remain in the validator set, running multiple validators will often result in a higher return than running a single validator.
```
```suggestion
It is possible for a single entity to run multiple validators.  Examples will show that it sometimes makes good business sense to do so.  Assuming you have enough DOTs, or enough stake nominates your validator, to ensure that your validators remain in the validator set, running multiple validators will often result in a higher return than running a single validator.
```

I push back a bit on the idea that the protocol explicitly encourages single entities to run multiple validators. It encourages a validator set not controlled by massive token holders, ideally a more decentralized validator set. IMO encouraging a single entity to run more than one validator doesn't seem sustainable long term, although I acknowledge it sometimes makes good business sense. 

Devil's Advocate: Why doesn't a large stakeholder simply nominate their tokens to other validators? It would be less costs on them and easier management.

jplatts12 replied:
Then they do not get the commissions, so they will make materially less rewards while also entrusting someone else with their tokens. If you have a validator set-up then spinning up another instance is a marginal cost.

laboon replied:
From a game-theoretical perspective, large stakeholders (large enough where splitting their stake to multiple validators allows all of the "baby validators" to remain in the validator set)  _are_ explicitly encouraged since it will mean increasing their share of the payout.  It also biases the system toward relatively equal stake between validators, as you want to stake just enough to ensure that your validator(s) stay(s) in the validator set.

laboon replied:
I'd be happy to say something like "are encouraged under certain circumstances" because I guess it's not actually "explicit" _per se_.

laboon replied:
Reworded to explain the benefits without actually saying that it is explicitly encouraged

lsaether commented:
Sometimes CI breaks for no good reason. When this happen you can usually fix it by restarting the job once or twice.

laboon commented:
> I would also suggest to include some information on validator preferences. Validators can choose to take a cut out of the rewards before distributing the payment to nominators and self stake. This is another incentive for a single entity to run more than one validator but does not get mentioned here.

Yes, I was trying to keep this as simple as possible (since all of this information _is_ already available on the wiki, it's just spread around a bit).  I'll detail it, although I'll limit to a paragraph and explain that for simplicity's sake, I am assuming no validator cut in the payout examples.

laboon commented:
This turned out to be more than just a few paragraphs as originally asked for! =)

I added a section at the bottom on nominating vs validating and how it impacts rewards.  I tried putting it earlier but I couldn't find a good place to make it flow.

I also added it to mkdocs.yml so that it is visible without going through the validator overview page.

@lsaether Can you take another look please?

laboon commented:
👍

laboon commented:
CC @jplatts12 , this is merged now, should be up on site soon

lsaether commented:
It does serve a purpose for being a single collection of all the links, but historically this was from before wiki times when it was used as the only central guide post.
